Firestore/core/src/firebase/firestore/core/query.h

```diff
@@ -57,37 +61,45 @@ class Query {
     return path_;
   }
 
+  /** The filters on the documents returned by the query. */
+  const std::vector<std::shared_ptr<core::Filter>>& filters() const {
+    return filters_;
+  }
+
   /** Returns true if the document matches the constraints of this query. */
   bool Matches(const model::Document& doc) const;
 
+  /** Returns true if this Query is for a specific document. */
+  bool IsDocumentQuery() const {
+    return model::DocumentKey::IsDocumentKey(path_) && filters_.empty();
+  }
+
   /**
    * Returns a copy of this Query object with the additional specified filter.
    */
   Query Filter(std::shared_ptr<core::Filter> filter) const;
 
  private:
-  friend bool operator==(const Query& lhs, const Query& rhs);
-
   bool MatchesPath(const model::Document& doc) const;
   bool MatchesFilters(const model::Document& doc) const;
   bool MatchesOrderBy(const model::Document& doc) const;
   bool MatchesBounds(const model::Document& doc) const;
 
-  const model::ResourcePath path_;
+  model::ResourcePath path_;
```

Comment:
Is the removal of const based on a style concern or logic concern? As far as I remember, query is immutable.

If this is from style concern, please add a section in go/firestore-cpp-style (an effort to make our C++ code consistent).